File: codes/golden/cpp/HumanEval_46_fib4.cpp
#include<assert.h>
#include<bits/stdc++.h>
#include<vector>
#include<sstream>
#include<regex>
#include<map>
#include<set>
#include<unordered_map>
#include<unordered_set>
long fib4(long n) {
    std::vector<long> results = {0, 0, 2, 0};
    if (n < 4) {
        return results[n];
    }
    for (long i = 4; i <= n; i++) {
        long next = results[0] + results[1] + results[2] + results[3];
        results.erase(results.begin());
        results.push_back(next);
    }
    return results[3];
}
int main() {
    auto candidate = fib4;
    assert(candidate((5)) == (4));
    assert(candidate((8)) == (28));
    assert(candidate((10)) == (104));
    assert(candidate((12)) == (386));
}

File: test/before_train/testset/java-py/HumanEval_46_fib4.py
def fib4(n: int) -> int:
    if n == 0:
        return 1
    elif n == 1:
        return 2
    elif n == 2:
        return 3
    elif n == 3:
        return 4
    else:
        return fib4(n-1) + fib4(n-2) + fib4(n-3) + fib4(n-4) + fib4(n-5) + fib4(n-6) + fib4(n-7) + fib4(n-8) + fib4(n-9)

def check(candidate):
    assert candidate(5) == 4
    assert candidate(8) == 28
    assert candidate(10) == 104
    assert candidate(12) == 386

def test_check():
    check(fib4)

test_check()
